src/main/java/com/pusher/client/channel/impl/ChannelImpl.java

```diff
@@ -111,18 +110,19 @@ public void onMessage(final String event, final String message) {
             if (listeners != null) {
                 for (final SubscriptionEventListener listener : listeners) {
                     final String data = extractDataFrom(message);
```

Comment:
This should be removed, along with the `extractDataFrom` method (sorry I thought you had already done this).

Reply:
done

Reply:
You should also remove the method itself:

https://github.com/pusher/pusher-websocket-java/blob/e9cc8250015376dbe297d4372a4ac169dddb4195/src/main/java/com/pusher/client/channel/impl/ChannelImpl.java#L191-L195